libs/community/langchain_community/vectorstores/opensearch_vector_search.py

```diff
@@ -840,27 +851,48 @@ def similarity_search_with_score(
         score_threshold: Optional[float] = 0.0,
         **kwargs: Any,
     ) -> List[Tuple[Document, float]]:
-        """Return docs and it's scores most similar to query.
-
-        By default, supports Approximate Search.
-        Also supports Script Scoring and Painless Scripting.
+        """Return docs and their similarity scores based on the input query.
 
         Args:
-            query: Text to look up documents similar to.
+            query: The query text to search for.
             k: Number of Documents to return. Defaults to 4.
             score_threshold: Specify a score threshold to return only documents
             above the threshold. Defaults to 0.0.
 
         Returns:
-            List of Documents along with its scores most similar to the query.
+            List of tuples containing the Document and its similarity score.
 
         Optional Args:
-            same as `similarity_search`
+            search_type: The type of search to perform. Can be one of:
+                - "approximate_search" (default)
+                - "script_scoring"
+                - "painless_scripting"
+                - "hybrid_search"
+            query_text: The query text to use for keyword search in hybrid search.
+            Other optional arguments are the same as `similarity_search`.
         """
         embedding = self.embedding_function.embed_query(query)
-        return self.similarity_search_with_score_by_vector(
-            embedding, k, score_threshold, **kwargs
+        hits = self._raw_similarity_search_with_score_by_vector(
+            embedding=embedding, k=k, query_text=query, **kwargs
         )
```

Comment:
What is the motivation for no longer delegating to `similarity_search_with_score_by_vector` here?